pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -105,6 +105,9 @@ func NewEndpointController(ctx context.Context, podInformer coreinformers.PodInf
 	e.podsSynced = podInformer.Informer().HasSynced
 
 	endpointsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, cur interface{}) {
```

Comment:
I found this change would break a conformance test: ["should test the lifecycle of an Endpoint"](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125675/pull-kubernetes-e2e-gce/1805301677411012608).
However, the conformance test itself seems contradictory with some logic of endpoints controller:
* The conformance test expects an Endpoint created without a Service to exist
* The `checkLeftoverEndpoints` function of endpoint controller triggers syncing all Endpoints except for those having leader election annotation and will cause Endpoints without Services to be deleted: https://github.com/kubernetes/kubernetes/blob/921fb0b7c5850db9b109f28dd669b81b1522e055/pkg/controller/endpoint/endpoints_controller.go#L561-L567

The conformance test gives an impression that an user created Endpoints (without a Service) can exist but it will be deleted once kube-controller-manager restarts or fails over.

Reply:
It seems the conformance test would fail as long as the controller watches endpoints update events.

I'm thinking another fix which doesn't affect the conformance test and could avoid doubling the work: we track the stale resource versions of endpoints after each successful endpoints update, use it to determine whether the endpoints in cache is stale when syncing it next time, and return error if it's stale and retry. (The reason why we don't track generation like endpointslice controller is endpoints doesn't have a generation.)